devices/disk-bios/disk-bios.c

```diff
@@ -236,8 +233,7 @@ static ssize_t diskbios_read(unsigned int minor, addr_t offs, void *buff, size_t
 
 		/* Read data from cache */
 		size = (sb == eb) ? len - l : SIZE_BLOCK - offs % SIZE_BLOCK;
-		hal_memcpy(buff, diskbios_common.rcache + s % BLOCKS_RCACHE * SIZE_BLOCK + offs % SIZE_BLOCK, size);
-		buff += size;
+		hal_memcpy((char *)buff + l, rcache + s % BLOCKS_RCACHE * SIZE_BLOCK + offs % SIZE_BLOCK, size);
```

Comment:
I think that brackets are needed here - intended operation order isn't clear

Reply:
Fixed.